support-frontend/app/admin/settings/Switches.scala

```diff
@@ -1,21 +1,121 @@
 package admin.settings
 
 import com.gu.support.encoding.Codec
-import com.gu.support.encoding.Codec.deriveCodec
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.auto._
+import io.circe.{Decoder, Encoder, Json}
+import io.circe.optics.JsonPath._
+
+sealed trait SwitchState {
+  def isOn: Boolean
+}
+case object On extends SwitchState {
+  def isOn: Boolean = true
+}
+case object Off extends SwitchState {
+  def isOn: Boolean = false
+}
```

Comment:
could even simplify further, assuming deriveEnumerationDecoder can handle it

```suggestion
sealed abstract class SwitchState(val isOn: Boolean)
case object On extends SwitchState(true)
case object Off extends SwitchState(false)
```

Reply:
Nice, it still works!